### ISSUE TITLE
AmazonPay inconsistent order number

### ISSUE THREAD

**Issue by mfritsche-dotsource** [PAYONE-GmbH/shopware-5#158]
Hello,

i noticed some flaw in the Payone AmazonPay order handling which is done in Controllers/MoptPaymentAmazon::finishAction(). 

At the beginning of the function the order number is reserved for transmission to Amazon:

```
        // pre-reserve Shop Order Number
        if (empty($this->session['moptAmazonOrdernum'])){
            $my_sOrder = new sOrder();
            $this->session['moptAmazonOrdernum'] = $my_sOrder->sGetOrderNumber();
        }
```
Later on when the process was successful, the order is saved and therefor a new order number created:

```
            $orderNumber = $this->saveOrder(
                $txid,
                $this->session->paymentReference
            );
```

Afterwards the newly created order number is overwritten by the one already sent to Amazon:

```
            // replace the new order number with the already reserved one, which is already sent to amazon
            $sql = 'UPDATE  `s_order` SET ordernumber = ? WHERE transactionID = ?';
            Shopware()->Db()->query($sql, array($this->session['moptAmazonOrdernum'], $txid));
```
This happens, as seen, only for the s_order table but not for the s_order_details table. Meaning that in the s_order table the order number sent to Amazon is used but in the s_order_details table it is still the one set by saveOrder. Creating two different order numbers for the same orderID.

Is that behaviour intentional? And if yes, whats the reasoning behind?

Best regards
Marco Fritsche

**Comment by fjbender** [comment]
I think this is a possible bug. We'll investigate.

Cheers
Florian

**Comment by fjbender** [comment]
Fixed with ed1f9b22af1bc84e125338b51b74afe35351b1fa
